src/Microsoft.Azure.WebJobs.Host/Scale/ScaleOptions.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Host.Scale
+{
+    /// <summary>
+    /// Configuration options for scaling.
+    /// </summary>
+    public class ScaleOptions
+    {
+        private TimeSpan _scaleMetricsMaxAge;
+        private TimeSpan _scaleMetricsSampleInterval;
+
+        public ScaleOptions()
+        {
+            // At the default values, a single monitor will be generating 6 samples per minute
+            // so at 2 minutes that's 12 samples
+            // Assume a case of 100 functions in an app, each mapping to a monitor. Thats
+            // 1200 samples to read from storage on each scale status request.
+            ScaleMetricsMaxAge = TimeSpan.FromMinutes(2);
+            ScaleMetricsSampleInterval = TimeSpan.FromSeconds(10);
+            MetricsPurgeEnabled = true;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating the maximum age for metrics.
+        /// Metrics that exceed this age will not be returned to monitors.
+        /// </summary>
+        public TimeSpan ScaleMetricsMaxAge
+        {
+            get
+            {
+                return _scaleMetricsMaxAge;
+            }
+
+            set
+            {
+                if (value < TimeSpan.FromMinutes(1) || value > TimeSpan.FromMinutes(5))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(ScaleMetricsMaxAge));
+                }
+                _scaleMetricsMaxAge = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the sampling interval for metrics.
+        /// </summary>
+        public TimeSpan ScaleMetricsSampleInterval
+        {
+            get
+            {
+                return _scaleMetricsSampleInterval;
+            }
+
+            set
+            {
+                if (value < TimeSpan.FromSeconds(1) || value > TimeSpan.FromSeconds(30))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(ScaleMetricsSampleInterval));
+                }
+                _scaleMetricsSampleInterval = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether old metrics data
+        /// will be auto purged.
+        /// </summary>
+        public bool MetricsPurgeEnabled { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether target base scaling is enabled at the host level.
+        /// </summary>
+        public bool IsTargetScalingEnabled { get; set; }
```

Comment:
To confirm, in Functions this is currently controlled by app setting TARGET_BASED_SCALING_ENABLED. Remind me again why we're translating this to an options? Just to allow additional configuration capabilities?

Reply:
Yes, there are 2 levels on enabling TBS:
1. App setting.
2. Specific trigger level.

BTW we made TBS default strategy recently (not released yet):
https://github.com/Azure/azure-functions-host/commit/e65a0a179fe957f084652b7a8e0b29a74253c4dc